### ISSUE TITLE
Please explain why ++ and -- operators are prohibited

### ISSUE THREAD

**Issue by golszewski86** [teamed/qulice#942]
Please see qulice-checkstyle/src/main/resources/com/qulice/checkstyle/checks.xml
```
<module name="IllegalToken">
   <property name="tokens" value="POST_INC, POST_DEC"/>
</module>
```
It prohibits usage of the post increment and decrement operators(like idx++ or idx--), however there is no justification for it. I wasn't able to find one on the intenet except people saying it's more efficient in C/C++.

Could you please provide justification, and maybe include it in the checks.xml file?

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) please, pay attention to this issue

**Comment by 0crat** [comment]
@golszewski86[/z](https://www.0crat.com/u/golszewski86) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C3T49A35L) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by krzyk** [comment]
@golszewski86 Yeah, you are right. And in most cases many devs hack it by using ++idx and --idx.

@yegor256 Can this rule go away? Or is there a justification for it?

**Comment by llorllale** [comment]
@krzyk a possible reason is to avoid this mistake:

```java
    int x = 1;
    int y = x++;
    System.out.println(x);
    System.out.println(y);  // someone may mistakenly expect y = 2
```

**Comment by golszewski86** [comment]
I'm sorry @llorllale but this reason doesn't make sense to me. I understand some people may not know how do the post-increment and pre-increment work, but this is not the reason to disable these constructions for everyone.

**Comment by krzyk** [comment]
@yegor256 yeah, but both postfix and prefix increment /decrement in my opinion is equally bad. So why discourage just one? (or at all?). I think we should either prohibit both or none.

**Comment by llorllale** [comment]
@krzyk why do you think the preincrement is bad?

**Comment by yegor256** [comment]
@krzyk I don't think that pre-increment is bad. Post-increment is bad for a very specific reason -- it returns the value before the increment, which makes code less readable. Pre-increment is side-effect free, so to speak. So, I vote for `++i` but against `i++`.

**Comment by golszewski86** [comment]
> Post-increment is bad for a very specific reason -- it returns the value before the increment, which makes code less readable
That's just an opinion @yegor256, maybe it's less readable for you but not for everyone. I'd expect code quality rules to be based on generally acknowledged rules, not personal opinions about readability.

**Comment by golszewski86** [comment]
@krzyk yep there are two arguments:
- Someone may not know how does post increment work and use it in a wrong way. However, it's not an argument against post increment operator but against literally any language functionality. Should we prohibit usage of TreeSet because some people don't know how it works?
- Yegor's opinion that post increment isn't readable. It's not an argument however but just an opinion. Is ++i really more readable than i++? Is there any objective metric of readiness?

**Comment by yegor256** [comment]
@ikarsokolov we're not planning to fix this, so I'm closing
